src/Mod/Spreadsheet/TestSpreadsheet.py

```diff
@@ -33,354 +33,561 @@
+        cls.sheet.set("E11", "=floor(3.4mm)")
+        cls.sheet.set("F11", "=floor(3.6mm)")
+        cls.sheet.set("G11", "=floor(-3.4mm)")
+        cls.sheet.set("H11", "=floor(-3.6mm)")
+        cls.sheet.set("A12", "=asin(0.5)")  # Asin
+        cls.sheet.set("B12", "=asin(0.5mm)")
+        cls.sheet.set("A13", "=acos(0.5)")  # Acos
+        cls.sheet.set("B13", "=acos(0.5mm)")
+        cls.sheet.set("A14", "=atan(sqrt(3))")  # Atan
+        cls.sheet.set("B14", "=atan(0.5mm)")
+        cls.sheet.set("A15", "=sinh(0.5)")  # Sinh
+        cls.sheet.set("B15", "=sinh(0.5mm)")
+        cls.sheet.set("A16", "=cosh(0.5)")  # Cosh
+        cls.sheet.set("B16", "=cosh(0.5mm)")
+        cls.sheet.set("A17", "=tanh(0.5)")  # Tanh
+        cls.sheet.set("B17", "=tanh(0.5mm)")
+        cls.sheet.set("A18", "=sqrt(4)")  # Sqrt
+        cls.sheet.set("B18", "=sqrt(4mm^2)")
+        cls.sheet.set("A19", "=mod(7; 4)")  # Mod
+        cls.sheet.set("B19", "=mod(-7; 4)")
+        cls.sheet.set("C19", "=mod(7mm; 4)")
+        cls.sheet.set("D19", "=mod(7mm; 4mm)")
+        cls.sheet.set("A20", "=atan2(3; 3)")  # Atan2
+        cls.sheet.set("B20", "=atan2(-3; 3)")
+        cls.sheet.set("C20", "=atan2(3mm; 3)")
+        cls.sheet.set("D20", "=atan2(3mm; 3mm)")
+        cls.sheet.set("A21", "=pow(7; 4)")  # Pow
+        cls.sheet.set("B21", "=pow(-7; 4)")
+        cls.sheet.set("C21", "=pow(7mm; 4)")
+        cls.sheet.set("D21", "=pow(7mm; 4mm)")
+        cls.sheet.set("A23", "=hypot(3; 4)")  # Hypot
+        cls.sheet.set("B23", "=hypot(-3; 4)")
+        cls.sheet.set("C23", "=hypot(3mm; 4)")
+        cls.sheet.set("D23", "=hypot(3mm; 4mm)")
+        cls.sheet.set("A24", "=hypot(3; 4; 5)")  # Hypot
+        cls.sheet.set("B24", "=hypot(-3; 4; 5)")
+        cls.sheet.set("C24", "=hypot(3mm; 4; 5)")
+        cls.sheet.set("D24", "=hypot(3mm; 4mm; 5mm)")
+        cls.sheet.set("A26", "=cath(5; 3)")  # Cath
+        cls.sheet.set("B26", "=cath(-5; 3)")
+        cls.sheet.set("C26", "=cath(5mm; 3)")
+        cls.sheet.set("D26", "=cath(5mm; 3mm)")
+
+        l = math.sqrt(5 * 5 + 4 * 4 + 3 * 3)
+        cls.sheet.set("A27", "=cath(%0.15f; 5; 4)" % l)  # Cath
+        cls.sheet.set("B27", "=cath(%0.15f; -5; 4)" % l)
+        cls.sheet.set("C27", "=cath(%0.15f mm; 5mm; 4)" % l)
+        cls.sheet.set("D27", "=cath(%0.15f mm; 5mm; 4mm)" % l)
+
+        cls.doc.recompute()
+
+    @classmethod
+    def tearDownClass(cls):
+        FreeCAD.closeDocument(cls.doc.Name)
+
     def assertMostlyEqual(self, a, b):
         if type(a) is Units.Quantity:
             self.assertTrue(math.fabs(a.Value - b.Value) < 1e-14)
             self.assertTrue(a.Unit == b.Unit)
         else:
+            self.assertNotIsInstance(a, str)
+            self.assertNotIsInstance(b, str)
+            self.assertTrue(
+                math.fabs(a - b) < 1e-14,
+                "Values are not equal: %s != %s" % (a, b),
+            )
             self.assertTrue(math.fabs(a - b) < 1e-14)
```

Comment:
You have the same assert twice.
```suggestion
            self.assertTrue(
                math.fabs(a - b) < 1e-14,
                "Values are not equal: %s != %s" % (a, b),
            )
```

We doesn't test for equality so `msg` should be adjusted to reflect what we test:
```suggestion
            self.assertTrue(
                math.fabs(a - b) < 1e-14,
                f"Values are not close enough: {a} and {b} has a diff of {math.fabs(a - b)}",
            )
```